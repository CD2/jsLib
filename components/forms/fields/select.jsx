import React from 'react'
import PropTypes from 'prop-types'

export class SelectField extends React.Component {

  static propTypes = {
    choices: PropTypes.arrayOf(PropTypes.object),
    default_value: PropTypes.string,
    disabled: PropTypes.bool,
    includeBlank: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    name: PropTypes.string,
    onChange: PropTypes.func,
    onRawChange: PropTypes.func,
    value: PropTypes.PropTypes.any,
  }

  static defaultProps = {
    choices: [],
    disable: false,
    value: ``,
  }

  handleChange = (e) => {
    const { onRawChange, onChange } = this.props
    if (onRawChange) onRawChange(e)
    if (onChange) onChange({ name: e.target.name, value: e.target.value })
  }

  renderChoices() {
    const { choices, includeBlank } = this.props
    const choiceHtml = []
    if (includeBlank) {
      const text = typeof includeBlank === `string` ? includeBlank : `--select--`
      choiceHtml.push(<option key="$BLANK$" value="">{text}</option>)
    }
    choices.forEach(choice => {
      let text
      let value
      if (typeof choice === `string`) {
        text = choice
        value = choice
      } else {
        text = choice.text
        value = choice.value
      }
      if (this.props.default_value === value){
        choiceHtml.push(<option key={value} value={value} selected="selected">{text}</option>)
      } else {
        choiceHtml.push(<option key={value} value={value}>{text}</option>)
      }
    })
    return choiceHtml
  }

  render() {
    const { name, value, disabled } = this.props
    return (
      <select
        name={name}
        className="select"
        value={value}
        disabled={disabled}
        onChange={this.handleChange}
      >
        {this.renderChoices()}
      </select>
    )
  }

}
export default SelectField
