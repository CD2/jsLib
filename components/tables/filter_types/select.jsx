import React from 'react'
import PropTypes from 'prop-types'

export default class ColumnFilterSearch extends React.Component {

  static propTypes = {
    name: PropTypes.string,
    options: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string,
      value: PropTypes.string,
    }))
  }

  state = {
    values: [],
  }

  handleChange = (e) => {
    const { name, checked } = e.target
    this.setState(prevState => {
      let values = (checked)
        ? [...prevState.values, name]
        : prevState.values.filter(v=> v !== name)
      return { values }
    })
  }

  handleSubmit = () => {
    this.props.onChange(this.state.value)
  }

  isChecked(value) {
    return this.state.values.includes(value)
  }

  renderOption(option) {
    return (
      <label>
        <input
          type='checkbox'
          name={option.value}
          checked={this.isChecked(option.value)}
          onChange={this.handleChange}/>
        {option.label}
      </label>
    )
  }

  render() {
    const { name, options } = this.props
    return (
      <div>
        <h4>{name}</h4>
        {options.map(option => this.renderOption(option))}
        <button onClick={this.handleSubmit}>Apply</button>
      </div>
    )
  }

}
