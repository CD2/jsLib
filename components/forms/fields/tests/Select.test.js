/* eslint-disable react/jsx-filename-extension */
import React from "react"
import { SelectField } from "../select"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"

describe(`<SelectField />`, () => {
  const view = shallow(<SelectField />)

  it(`snapshot`, () => expect(toJson(view)).toMatchSnapshot())
})
