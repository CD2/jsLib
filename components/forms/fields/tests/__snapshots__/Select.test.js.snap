// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectField /> snapshot 1`] = `
<select
  className="select"
  onChange={[Function]}
  value=""
/>
`;
